### PR TITLE
Amend service recipients

### PR DESCRIPTION
**Pull request by magnusjohn-bjss**

### CODE REVIEW

**Review by imranbashir01 — APPROVED** [review]

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Ordering/SolutionSelection/AdditionalServices/SelectAdditionalServiceRecipients.cs [hunk 1]
```diff
@@ -102,10 +102,10 @@ public void SelectAdditionalServiceRecipients_SelectionMade_ExpectedResult()
             CommonActions.ClickSave();
 
             CommonActions.PageLoadedCorrectGetIndex(
-                typeof(PricesController),
-                nameof(PricesController.SelectPrice)).Should().BeTrue();
+                typeof(ServiceRecipientsController),
+                nameof(ServiceRecipientsController.ConfirmChanges)).Should().BeTrue();
 
-            GetOrderItem().OrderItemRecipients.Count.Should().Be(1);
+            GetOrderItem().OrderItemRecipients.Count.Should().Be(0);
```
Comment (ghost) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Ordering/SolutionSelection/AdditionalServices/SelectAdditionalServiceRecipients.cs:108]: `Count.Should().Be(0)` -> `.Count.Should().BeEmpty();`

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Ordering/SolutionSelection/AdditionalServices/ConfirmAdditionalServiceRecipientsAmendment.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                typeof(ServiceRecipientsController),
+                nameof(ServiceRecipientsController.EditServiceRecipients)).Should().BeTrue();
+
+            CommonActions.GetNumberOfSelectedCheckBoxes().Should().Be(2);
+        }
+
+        [Fact]
+        public void ConfirmAdditionalServiceRecipientsAmendment_ClickContinue_ExpectedResult()
+        {
+            GetService().OrderItemRecipients.Count.Should().Be(0);
```
Comment (ghost) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Ordering/SolutionSelection/AdditionalServices/ConfirmAdditionalServiceRecipientsAmendment.cs:85]: `Count.Should().Be(0)` -> `Count.Should().BeEmpty()`

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Ordering/SolutionSelection/AdditionalServices/ConfirmAdditionalServiceRecipients.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                typeof(ServiceRecipientsController),
+                nameof(ServiceRecipientsController.AddServiceRecipients)).Should().BeTrue();
+
+            CommonActions.GetNumberOfSelectedCheckBoxes().Should().Be(2);
+        }
+
+        [Fact]
+        public void ConfirmAdditionalServiceRecipients_ClickContinue_ExpectedResult()
+        {
+            GetService().OrderItemRecipients.Count.Should().Be(0);
```
Comment (ghost) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Ordering/SolutionSelection/AdditionalServices/ConfirmAdditionalServiceRecipients.cs:85]: `Count.Should().Be(0)` -> `Count.Should().BeEmpty()`

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Orders/Controllers/SolutionSelection/AdditionalServicesController.cs [hunk 4]
```diff
@@ -216,6 +211,27 @@ public async Task<IActionResult> ConfirmAdditionalServiceChanges(string internal
                 new { internalOrgId, callOffId, model.ToAdd.First().CatalogueItemId });
         }
 
+        private static List<CatalogueItemId> GetCurrentServiceIds(CallOffId callOffId, OrderWrapper wrapper)
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Orders/Controllers/SolutionSelection/AdditionalServicesController.cs:214]: For the sake of Codecov, it's worth adding some tests for this. Seems like only 216-222 are being tested indirectly via the calling member.

**Review by bjss-colin — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Orders/Controllers/SolutionSelection/AdditionalServicesController.cs (hunk 4 above)
Comment (magnusjohn-bjss) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Orders/Controllers/SolutionSelection/AdditionalServicesController.cs:214]: Given this is gonna need refactoring and these tests are a massive pain to setup, any chance we can look the other way on this one and go for 100% test coverage on the refactored service code? we'll need to refactor before the amend stuff goes live as there's an additional "copy service recipients" screen coming in, this controller code is waaaaaay too complex and brittle now to extend as-is

**Review by bjss-colin — APPROVED** [review]

**Pull request merged**